openquake/hazardlib/gsim/tromans_2019.py

```diff
@@ -0,0 +1,364 @@
+    IMT   const_phiss   phi1m  phi2m
+    pgv          0.46    0.49   0.37
+    pga          0.46    0.49   0.35
+    0.01         0.46    0.49   0.35
+    0.1          0.45    0.45   0.43
+    0.2          0.48    0.51   0.37
+    0.3          0.48    0.51   0.37
+    0.5          0.46    0.49   0.37
+    1.0          0.45    0.46   0.40
+    3.0          0.41    0.41   0.41
+    """)
+
+
+#: Single ststion dfS2S model
+DELTA_PHI_S2S = CoeffsTable(sa_damping=5, table="""\
+    IMT          dfs2s
+    pga         0.0000
+    0.01000     0.0000
+    0.99999     0.0000
+    1.00000     0.0001
+    2.00000     0.0069
+    3.00000     0.0131
+    10.0000     0.0131
+    """)
+
+
+def get_heteroskedastic_tau(imt, mag):
+    """
+    Returns the magnitude dependent inter-event variability using the
+    model of Abrahamson et al (2014).
+    :param dict C:
+        Coefficients dictionary
+    :param float mag:
+        Magnitude
+    """
+
+    C = ASK_TAU_COEFFS[imt]
+    if mag < 5:
+        tau = C['s3']
+    elif mag <= 7:
+        tau = C['s3'] + (C['s4'] - C['s3']) / 2. * (mag - 5.)
+    else:
+        tau = C['s4']
+    return tau
+
+
+def get_heteroskedastic_phi(imt, mag):
+    """
+    Returns the heteroskedastic intra-event term, taken as the maximum
+    of the constant single-station phi and the magnitude dependent
+    single-station phi provided in Table 4 of Rodriguez-Marek et al (2014)
+    """
+    C = PHI_SS_COEFFS[imt]
+    if mag < 5.0:
+        mag_phi = C["phi1m"]
+    elif mag > 7.0:
+        mag_phi = C["phi2m"]
+    else:
+        mag_phi = C["phi1m"] + (C["phi2m"] - C["phi1m"]) * ((mag - 5.0) / 2.0)
+    if mag_phi > C["const_phiss"]:
+        return mag_phi
+    else:
+        return C["const_phiss"]
+
+
+HETEROSKEDASTIC_PHI = {
+    "upper": lambda imt, mag: 1.16 * get_heteroskedastic_phi(imt, mag),
+    "central": lambda imt, mag: get_heteroskedastic_phi(imt, mag),
+    "lower": lambda imt, mag: 0.84 * get_heteroskedastic_phi(imt, mag)}
+
+HOMOSKEDASTIC_PHI = {
+    "upper": lambda imt: 1.16 * PHI_SS_COEFFS[imt]["const_phiss"],
+    "central": lambda imt: PHI_SS_COEFFS[imt]["const_phiss"],
+    "lower": lambda imt: 0.84 * PHI_SS_COEFFS[imt]["const_phiss"]}
+
+
+HETEROSKEDASTIC_TAU = {
+    "upper": lambda imt, mag: get_heteroskedastic_tau(imt, mag) + 0.075,
+    "central": lambda imt, mag: get_heteroskedastic_tau(imt, mag),
+    "lower": lambda imt, mag: get_heteroskedastic_tau(imt, mag) - 0.075}
+
+HOMOSKEDASTIC_TAU = {
+    "upper": lambda imt: get_heteroskedastic_tau(imt, 6.0) + 0.075,
+    "central": lambda imt: get_heteroskedastic_tau(imt, 6.0),
+    "lower": lambda imt: get_heteroskedastic_tau(imt, 6.0) - 0.075}
+
+
+uppernames = '''
+DEFINED_FOR_INTENSITY_MEASURE_TYPES
+DEFINED_FOR_STANDARD_DEVIATION_TYPES
+REQUIRES_SITES_PARAMETERS
+REQUIRES_RUPTURE_PARAMETERS
+REQUIRES_DISTANCES
+'''.split()
+
+
+class TromansEtAl2019(GMPE):
+    """
+    Implements a meta GMPE to apply the standard deviation model and
```

Comment:
We decided to change the name for "meta GMPE" to "modified GMPE" to avoid confusions with the MetaGSIM class.